src/Orleans/Storage/DbExtensions.cs

```diff
@@ -0,0 +1,192 @@
+        /// <remarks>This function throws if the given <see paramref="fieldName"/> does not exist.</remarks>
+        public static TValue GetValueOrDefault<TValue>(this IDataRecord record, string fieldName, TValue @default = default(TValue))
+        {
+            var ordinal = record.GetOrdinal(fieldName);
+            return record.IsDBNull(ordinal) ? @default : (TValue)record.GetValue(ordinal);
+        }
+
+
+        /// <summary>
+        /// Returns a value if it is not <see cref="System.DBNull"/>, <em>default(TValue)</em> otherwise.
+        /// </summary>
+        /// <typeparam name="TValue">The type of the value to request.</typeparam>
+        /// <param name="record">The record from which to retrieve the value.</param>
+        /// <param name="ordinal">The ordinal of the fieldname.</param>
+        /// <param name="@default">The default value if value in position is <see cref="System.DBNull"/>.</param>
+        /// <returns>Either the given value or the default for the requested type.</returns>
+        /// <exception cref="IndexOutOfRangeException"/>
+        /// <remarks>This function throws if the given <see paramref="fieldName"/> does not exist.</remarks>        
+        public static TValue GetValueOrDefault<TValue>(this IDataRecord record, int ordinal, TValue @default = default(TValue))
+        {
+            return record.IsDBNull(ordinal) ? @default : (TValue)record.GetValue(ordinal);
+        }
+
+
+        /// <summary>
+        /// Returns a value with the given <see paramref="fieldName"/>.
+        /// </summary>
+        /// <typeparam name="TValue">The type of value to retrieve.</typeparam>
+        /// <param name="record">The record from which to retrieve the value.</param>
+        /// <param name="fieldName">The name of the field.</param>
+        /// <returns>Value in the given field indicated by <see paramref="fieldName"/>.</returns>
+        /// <exception cref="IndexOutOfRangeException"/>
+        /// <remarks>This function throws if the given <see paramref="fieldName"/> does not exist.</remarks>        
+        public static TValue GetValue<TValue>(this IDataRecord record, string fieldName)
+        {
+            var ordinal = record.GetOrdinal(fieldName);
+            return (TValue)record.GetValue(ordinal);
+        }        
+        
+
+
+        /// <summary>
+        /// Adds given parameters to a command using reflection.
+        /// </summary>
+        /// <typeparam name="T">The type of the parameters.</typeparam>
+        /// <param name="command">The command.</param>
+        /// <param name="parameters">The parameters.</param>
+        /// <param name="nameMap">Maps a given property name to another one defined in the map.</param>
+        /// <remarks>Does not support collection parameters currently. Does not cache reflection results.</remarks>
+        public static void ReflectionParameterProvider<T>(this IDbCommand command, T parameters, IReadOnlyDictionary<string, string> nameMap = null)
```

Comment:
The following functions handle parameter or return value objects in queries by reflection and either read or fill in the values by reflection from the database by matching a name. Basically what other ORMs, such as Dapper, are doing.